### PR TITLE
[SIS-051] Blank Library Page

### LINKED ISSUE #51: [SIS-051] Blank Library Page
> As a user of the Sistema Lesson Planner, I want to be able to see a library of my lesson plans.

## Technical Information

Inside `/src/library/Library.js`:
- [x] Add background colours 
- [x] Add title "Lesson Plans" and a dummy menu button on the right (use flexbox to position these two)
- [x] Create a reusable component for a lesson plan button. Expand the button height if the title is super long, but don't push over the icons. Put the component in a new directory in `/src/library/components`
     - [x] Add dummy heart and menu button icons within this
- [x] Create a scrollable container to place the LP UI elements in (see `ScrollView`)

By dummy, I mean that `onPress` doesn't have to do anything right now. 

## Design

The Figma node for this screen can be found [here](https://www.figma.com/file/kSZduD0A5hk8sHmvqiC7DE/Sistema-Hi-Fidelity-Designs-%2B-Design-System?node-id=278%3A16162&t=uBWLplRiS9RKkpMh-0) in the wireframe and will give you info like font, colour hex codes, border-radius (for rounded corners), padding, etc.

image

## Good Resources

[View](https://reactnative.dev/docs/view)
[Stylesheet](https://reactnative.dev/docs/stylesheet)
[ScrollView](https://reactnative.dev/docs/scrollview/)
[TouchableOpacity](https://reactnative.dev/docs/touchableopacity)
[Flexbox](https://reactnative.dev/docs/flexbox) for centering
[SVG docs](https://github.com/software-mansion/react-native-svg/blob/main/USAGE.md) -- for icons
1. Go to the Figma [hi-fidelity wireframe](https://www.figma.com/file/kSZduD0A5hk8sHmvqiC7DE/Sistema-Hi-Fidelity-Designs-%2B-Design-System?node-id=42%3A12578) to export the icons as SVGs. On the right sidebar, Design > Export > SVG. 
2. See example usage of SVGs [here](https://github.com/uoftblueprint/sistema/blob/097f08271e46bce430453430d04719ecff60a771/packages/mobile/assets/HomeIcon.js). 
Although, check in with this [ticket](https://github.com/uoftblueprint/sistema/issues/47) first to see if a new, easier package can be used.

### PR DESCRIPTION
**Pull request by BakerWJ**
# Description

Create a blank Library page and a LessonPlanButton component.

image

Ticket #51 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My PR title is formatted as [SIS-\<issue number\>] \<issue name\> (eg. [SIS-010] Create UI button)
- [x] I have linked my issue/ticket on the project board to this PR
- [x] I have sent my PR to #team-sistema-prs on Slack and requested two people to review
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have tested that my feature works as expected on Android and iOS. If you are a Windows dev, please request a Mac dev to run your branch. If it's a front-end change, I have attached screenshots of what the result looks like on both platforms.

### CODE REVIEW

**Comment by BakerWJ** [comment]
Currently using RN 0.71 CSS flexbox gaps feature -- will likely create a different PR for upgrading RN version

**Review by harmitgoswami — CHANGES_REQUESTED** [review]
Is it possible to make the heart icons more in-line with the 3 dots?

**Comment by BakerWJ** [comment]
> Is it possible to make the heart icons more in-line with the 3 dots?

@harmitgoswami I could -- it seems like the issue is more on the SVG side though, as both are currently the same size and aligned with each other. Maybe it is possible that the exported SVGs aren't properly centered -- @yuemily24 thoughts on how this should be resolved?

**Comment by BakerWJ** [comment]
@harmitgoswami fixed -- I took this picture on a simulator running 0.70 which is why they aren't distanced properly, but they are vertically aligned now

image

**Review by harmitgoswami — APPROVED** [review]

**Comment by harmitgoswami** [comment]
lgtm now! mb for being so 🤓

**Comment by ramyzhang** [comment]
hmm is it also possible to center the sorting icon w the title header vertically as well? it looks a teensy bit off to me

**Comment by BakerWJ** [comment]
> hmm is it also possible to center the sorting icon w the title header vertically as well? it looks a teensy bit off to me

@ramyzhang maybe this is something we can bring up with the designer -- if we move over the button then it won't be aligned with the right side of the lesson plans :(. I'm not sure which is better, but I can slide it over if that is the preference

**Comment by ramyzhang** [comment]
> > hmm is it also possible to center the sorting icon w the title header vertically as well? it looks a teensy bit off to me
> 
> @ramyzhang maybe this is something we can bring up with the designer -- if we move over the button then it won't be aligned with the right side of the lesson plans :(. I'm not sure which is better, but I can slide it over if that is the preference

ohhhh i meant like moving the sorting button down just a bit so it looks like it's on the same line as the title. if not yeah no worries i'll bring it up w the designers and get back to u

**Comment by BakerWJ** [comment]
> ohhhh i meant like moving the sorting button down just a bit so it looks like it's on the same line as the title. if not yeah no worries i'll bring it up w the designers and get back to u

This should an easy fix -- I'll check the padding for the icon right now

**Comment by BakerWJ** [comment]
image

@ramyzhang is this centered im going insane and I can't tell

**Comment by ramyzhang** [comment]
@BakerWJ LOL yes it looks perfect thank you!!!

**Comment by ramyzhang** [comment]
@Lemiex pinging for reviewwww

File: src/Components/Header.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
   Text,
   TouchableOpacity
 } from 'react-native';
-import SistemaLogo from '../../assets/SistemaLogo.svg';
+import SistemaLogo from '../../assets/sistemaLogo.svg';
```
Comment (Lemiex) [inline src/Components/Header.js:9]: THANK YOU

**Review by Lemiex — APPROVED** [review]
Everything looks great!

File: src/library/Library.js [hunk 2]
```diff
@@ -1,19 +1,65 @@
...
+    fontFamily: 'Poppins-Bold',
+    fontSize: 28,
+    letterSpacing: 0.3,
+    color: '#453E3D'
+  },
+  inlineTitle: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    flexDirection: 'row',
+    marginHorizontal: 30,
```
Comment (ramyzhang) [inline src/library/Library.js:50]: hey i notice there aren't any units here but looking at the documentation didn't yield any info on what exactly the value would be treated as; some people said that some browsers take no units to be pixels. just to be safe, could you change this to %?

File: src/library/components/LessonPlanButton.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    fontFamily: 'Mulish-Regular',
+    fontSize: 16,
+    letterSpacing: 0.3,
+    color: 'black'
+  },
+  icons: {
+    display: 'flex',
+    flexDirection: 'row',
+    gap: 17,
+    margin: 14
```
Comment (ramyzhang) [inline src/library/components/LessonPlanButton.js:45]: same comment as the above for the margins here!

**Review by ramyzhang — CHANGES_REQUESTED** [review]
hey everything looks great (haven't had a chance to test the build bc my laptops being annoying), but i did leave some notes on the values for the margin styling. let me know ur thoughts!

File: src/library/Library.js (hunk 2 above)
Comment (BakerWJ) [reply src/library/Library.js:50]: React treats all numerical style values as px values if there is no unit. I'm not sure if it makes sense to convert this to a %  though -- shouldn't the margins stay the same regardless of the width / height of the device?

Comment (ramyzhang) [reply src/library/Library.js:50]: that's a good point, i'll merge this for now and if any issues come up later re: compatibility we'll revisit this

**Review by ramyzhang — APPROVED** [review]

**Pull request merged**